google/cloud/storage/internal/insert_object_media_request.h

```diff
@@ -29,10 +29,10 @@ namespace internal {
  * TODO(#710) - add missing request parameters.
  */
 class InsertObjectMediaRequest
-    : private RequestParameters<Generation, IfGenerationMatch,
-                                IfGenerationNotMatch, IfMetaGenerationMatch,
-                                IfMetaGenerationNotMatch, Projection,
-                                UserProject> {
+    : public GenericRequest<InsertObjectMediaRequest, Generation,
```

Comment:
It's both terrifying and amazing that C++ allows you to pass in the not-yet-defined class as a templated typename to the base class you're inheriting from. Neat.